cloudwatch_agent.tf

```diff
@@ -63,29 +63,36 @@ resource "aws_iam_role_policy_attachment" "cwagent-eks" {
 }
 
 locals {
-  # build a service account manifest map
-  service_account_manifest_templated = templatefile("${path.module}/yamls/cwagent-serviceaccount.yaml", {
-    account_id          = var.account_id,
-    cloudwatch_iam_role = aws_iam_role.cwagent-eks[0].name,
-  })
-  service_account_manifest_splitted  = split("---", local.service_account_manifest_templated)
-  service_account_manifest_list      = var.enable_cloudwatch_agent ? local.service_account_manifest_splitted : []
-  service_account_manifest_map       = {for mn in local.service_account_manifest_list : md5(mn) => mn}
-  # build deamonset manifest map
-  daemonset_manifest                 = file("${path.module}/yamls/cwagent-daemonset.yaml")
-  daemonset_manifest_splitted        = split("---", local.daemonset_manifest)
-  daemonset_manifest_list            = var.enable_cloudwatch_agent ? local.daemonset_manifest_splitted : []
-  daemonset_manifest_map             = {for mn in local.daemonset_manifest_list : md5(mn) => mn}
+  service_account_manifests = var.enable_cloudwatch_agent ? (
+    split("---", templatefile("${path.module}/yamls/cwagent-serviceaccount.yaml", {
+      account_id          = var.account_id
+      cloudwatch_iam_role = aws_iam_role.cwagent-eks[0].name
+    }))
+  ) : []
+
+  daemonset_manifests = var.enable_cloudwatch_agent ? (
+    split("---", file("${path.module}/yamls/cwagent-daemonset.yaml"))
+  ): []
```

Comment:
I the future I want to replace the `kubectl` provider with the kubernetes provider. The command to convert yaml -> tf was not something that either existed or was aware of at the time.